easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-7.0-foss-2021b.eb

```diff
@@ -0,0 +1,47 @@
+name = 'QuantumESPRESSO'
+version = '7.0'
+
+homepage = 'https://www.quantum-espresso.org'
+description = """Quantum ESPRESSO  is an integrated suite of computer codes
+for electronic-structure calculations and materials modeling at the nanoscale.
+It is based on density-functional theory, plane waves, and pseudopotentials
+(both norm-conserving and ultrasoft).
+"""
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+toolchainopts = {'usempi': True, 'openmp': True}
+
+source_urls = [
+    'https://github.com/QEF/q-e/releases/download/qe-%(version)s/',
+    'https://github.com/dceresoli/qe-gipaw/releases/download/%(version)s/',
+    'https://github.com/wannier-developers/wannier90/archive/'
+]
+sources = [
+    'qe-%(version)s-ReleasePack.tgz',
+    'qe-gipaw-%(version)s.tar.gz',
+    {'filename': 'wannier90-3.1.0.tar.gz', 'download_filename': 'v3.1.0.tar.gz'},
+]
+patches = ['QuantumESPRESSO-7.0-gipaw-string-preprocessor.patch']
```

Comment:
Is it worth reaching out to them about the in-place update, and kindly reminding them that this is really bad practice?

Reply:
I tried both with and without the current patch file, fails in both cases.

@zao So it seems like we'll need to redo the patch for the updated `qi-gipaw` sources?

Reply:
Currently, yes. The patch mismatches because upstream ran a full Intel preprocessor on the file resulting in removed ifdefs for OpenMP and expanding some lines past default line length.

Reply:
@zao So how did you create the patch? Mostly manually?

Reply:
Yeah, identifying the unsupported macros and evaluating them manually in Vim, splitting the resulting statements to avoid line length limitations.
It might be easier if we find a way to get a compiler flag in to accept over-long lines, but it still has to be more or less done manually.

Reply:
Looking at https://github.com/dceresoli/qe-gipaw/blob/7.0/src/xml_routines.f90 and the patch, I believe this has been fixed. Though there has been an inplace update to version 7.0 to do this.